### PR TITLE
feat(aio): enable previews for any PR

### PR DESCRIPTION
**Pull request by gkalpak**
This PR adds support for verifying PRs as "appropriate" for having their builds deployed to the preview server via a GitHub label (TBD). This will essentially enable having previews for any PR, regardless of the author, as long as its content has been manually reviewed and verified.

(This PR also includes some small refactorings, as well as dependencies and docs updates.)

### CODE REVIEW

**Comment by gkalpak** [comment]
I tried to break it up into smaller commits (for easier reviewing), but there are a couple of large-ish ones in there 😕 

Todo (in a follow-up PR):
- Expose the `changePrVisibility()` method through an API.
- Set up a GitHub webhook to automatically update a PR's verification status, when the specified label is added/removed.

File: aio/aio-builds-setup/dockerbuild/scripts-sh/health-check.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-set +e -o pipefail
+set +e -u -o pipefail
```
Comment (IgorMinar) [inline aio/aio-builds-setup/dockerbuild/scripts-sh/health-check.sh]: can you add a comment that some errors are expected and that's why we have `+e` here?

**Review by IgorMinar — APPROVED** [review]
Great work @gkalpak!

File: aio/aio-builds-setup/dockerbuild/scripts-js/lib/upload-server/upload-server-factory.ts [hunk 2]
```diff
@@ -56,12 +59,26 @@ class UploadServerFactory {
                                domainName: string): BuildCreator {
     const buildCreator = new BuildCreator(buildsDir);
     const githubPullRequests = new GithubPullRequests(githubToken, repoSlug);
+    const postPreviewsComment = (pr: number, shas: string[]) => {
+      const body = [
+        ...shas.map((sha, i) => `The angular.io preview for ${sha} is available [here][${i + 1}].`),
```
Comment (gkalpak) [inline aio/aio-builds-setup/dockerbuild/scripts-js/lib/upload-server/upload-server-factory.ts]: I went with:
> You can preview ${sha} at {url}.

File: aio/aio-builds-setup/dockerbuild/scripts-sh/health-check.sh (hunk 1 above)
Comment (gkalpak) [inline aio/aio-builds-setup/dockerbuild/scripts-sh/health-check.sh]: Done.

**Comment by gkalpak** [comment]
Addressed comments, fixed linting issue and rebased on master.
Waiting for Travis and will mark this for merge.

**Pull request merged**